File: 1544-make-the-string-great/1544-make-the-string-great.cpp
class Solution {
public:
    int getabs(char c){
        if(islower(c))
            return c-'a';
        return c-'A';
    }
    
    string makeGood(string s) {
        stack<char>st;
        
        for(char&c:s){
            if(st.size()==0){
                st.push(c);
            }
            else{
                if(st.top()!=c && getabs(st.top())==getabs(c))
                    st.pop();
                else
                    st.push(c);
            }
        }
        
        s="";
        while(st.size()){
            s+=st.top();
            st.pop();
        }
        reverse(s.begin(),s.end());
        return s;
    }
};

File: 1544-make-the-string-great/1544-make-the-string-great.java
class Solution {
    public String makeGood(String s) {
        Stack<Character>st=new Stack<>();
        int len=s.length();
        
        for(int i=0;i<len;++i){
            if(!st.isEmpty() && ((st.peek()+32)==s.charAt(i) || st.peek()==(s.charAt(i)+32))){
                st.pop();
            }
            else{
                st.push(s.charAt(i));
            }
        }
        
        StringBuilder sb=new StringBuilder();
        
        while(!st.empty()){
            sb.append(st.peek());
            st.pop();
        }
        
        return sb.reverse().toString();
    }
}